InformationScripting/src/queries/QueryExecutor.h

```diff
@@ -28,24 +28,34 @@
 
 #include "../informationscripting_api.h"
 
+#include "../misc/Optional.h"
+#include "TopLevelQuery.h"
+
 namespace Interaction {
 	class CommandResult;
 }
 
 namespace InformationScripting {
 
+class TopLevelQuery;
+class TupleSet;
 class Query;
 
 class INFORMATIONSCRIPTING_API QueryExecutor
 {
 	public:
-		QueryExecutor(Query* q);
 		~QueryExecutor();
+		void addQuery(std::unique_ptr<TopLevelQuery>&& query);
 
 		Interaction::CommandResult* execute();
 
+		QList<Optional<TupleSet> > runContinuation(const QList<TupleSet>& input);
```

Comment:
I'm not entirely very happy about the new API. Do we really need three methods for executing queries: execute, runContinuation, and executeLast? I'm especially not happy about having two public methods. Can't we reduce this to one?